### ISSUE TITLE
Expose `coderd_workspaces` as a Prometheus metric

### ISSUE THREAD

**Issue by bpmct** [coder/coder#12462]
We currently have workspace_builds, but being able to query the current number of workspaces (including their template and state) can be valuable for understanding usage. We should expose similar fields as the builds metric

```sh
# HELP coderd_workspace_builds_total The number of workspaces started, updated, or deleted.
# TYPE coderd_workspace_builds_total counter
coderd_workspace_builds_total{status="failed",template_name="docker",template_version="dazzling_chaum0",workspace_name="workspace-1",workspace_owner="admin",workspace_transition="START"} 1
coderd_workspace_builds_total{status="success",template_name="docker",template_version="dazzling_chaum0",workspace_name="workspace-1",workspace_owner="admin",workspace_transition="START"} 1
coderd_workspace_builds_total{status="success",template_name="docker",template_version="dazzling_chaum0",workspace_name="workspace-1",workspace_owner="admin",workspace_transition="STOP"} 1
```

**Comment by spikecurtis** [comment]
We can do this, but it has sharp edges around multiple Coderd replicas.

Essentially, the metric is a statistic that is scoped to the entire deployment (that is, workspaces aren't tied to a particular coderd server), but each Coderd would report the metric.  It would be very easy to write a dashboard that sums the metric, in which case you'll over-report by the multiple of Coderds you have.

**Comment by spikecurtis** [comment]
One other thing is that I re-read the original customer ask, and I think what they want is a GUAGE that tracks the current number of workspaces in a particular state, not a counter of builds.

**Comment by dannykopping** [comment]
Closed by #12762
